conanfile.py

```diff
@@ -459,13 +460,20 @@ def build(self):
                     args.append("\"%s_INCDIR=%s\"" % (var, self.deps_cpp_info[package].include_paths[-1]))
                 args += ["-D " + s for s in self.deps_cpp_info[package].defines]
 
+                def _remove_duplicate(l):
+                    seen = set()
+                    seen_add = seen.add
+                    for element in itertools.filterfalse(seen.__contains__, l):
+                        seen_add(element)
+                        yield element
+                
                 def _gather_libs(p):
                     libs = ["-l" + i for i in self.deps_cpp_info[p].libs]
                     libs += self.deps_cpp_info[p].sharedlinkflags
                     for dep in self.deps_cpp_info[p].public_deps:
                         libs += ["-L" + lpath for lpath in self.deps_cpp_info[dep].lib_paths]
                         libs += _gather_libs(dep)
-                    return libs
+                    return _remove_duplicate(libs)
```

Comment:
Is it somehow possible to replace this _remove_duplicate (which fixes msvc shared builds, great!) and use a python set (curly braces) in _gather_libs instead ?

Reply:
I try to return list(set(libs)), but element order is lost, and this could break the build with gcc and clang. So I use a simplified version on [unique_everseen](https://docs.python.org/3.5/library/itertools.html?highlight=unique_everseen#itertools-recipes)